core/Cargo.toml

```diff
@@ -21,7 +21,7 @@ std = ["math/std", "winter-utils/std"]
 
 [dependencies]
 math = { package = "winter-math", version = "0.6", default-features = false }
-crypto = { package = "miden-crypto", version = "0.3", default-features = false }
+crypto = { package = "miden-crypto", git = "https://github.com/0xPolygonMiden/crypto.git", branch = "next", default-features = false }
```

Comment:
The `MerkleStore` update has been merged, but not released. I suppose this PR needs to wait the next release of that crate prior to be merged. But the feature is done and the tests are green.